src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -173,6 +174,19 @@ std::string postQueryContext
 
   cleanPayload = xmlPayloadClean(out.c_str(), "<queryContextResponse>");
 
+  if ((cleanPayload == NULL) || (cleanPayload[0] == 0))
+  {
+    QueryContextResponse qcrs;
+
+    // FIXME P1: What StatusCode should be used here?
+    qcrs.errorCode.fill(SccReceiverInternalError, "invalid context provider response");
```

Comment:
I don't think this should be a 500 Internal Error. In general, fails in the connection with a CPr should be resolved as if there wasn't any CPr registered for the given context element, i.e. 404 Not Context Element found, from the perspective of the client. Of course, from an "internal" perspective it's perfectly ok (as you do some lines below) logging it as warning message in the log.